src/drivers/next-json-cookie/encrypt-cookie.ts

```diff
@@ -0,0 +1,18 @@
+import AES from 'crypto-js/aes';
+import Utf8 from 'crypto-js/enc-utf8';
+import JSONValue from '../types/json';
+import isJsonable from '../../support/is-jsonable';
+
+const COOKIE_ENCRYPTION_KEY = process.env.NEXT_PUBLIC_COOKIE_ENCRYPTION_KEY ?? 'COOKIE_ENCRYPTION_KEY';
+
+export const encryptCookie = <T = JSONValue>(value: T, crypto?: typeof AES, encoder?: typeof Utf8): string => {
+  const valueToEncrypt = (typeof value === 'string') ? value : JSON.stringify(value);
+  if (!crypto || !encoder) return valueToEncrypt;
+  return crypto.encrypt(valueToEncrypt, COOKIE_ENCRYPTION_KEY).toString();
+}
+export const decryptCookie = <T = JSONValue>(value: string, crypto?: typeof AES, encoder?: typeof Utf8): T => {
+  const valueToDecrypt = isJsonable<T>(value) ? JSON.parse(value) : value;
+  if (!crypto || !encoder) return valueToDecrypt;
+  const bytes = crypto.decrypt(valueToDecrypt, COOKIE_ENCRYPTION_KEY);
+  return JSON.parse(bytes.toString(encoder) ?? '{}');
+}
```

Comment:
Rather than doing a complicated check to see if a value is JSON-able, I find it easier to simply try/catch, if it is JSON-able, great! If not, oh well!
```suggestion
export const decryptCookie = <T = JSONValue>(value: string, crypto?: typeof AES, encoder?: typeof Utf8): T => {
  try {
    const valueToDecrypt = JSON.parse(value);
    
    if (!crypto || !encoder) return valueToDecrypt;
    
    const bytes = crypto.decrypt(valueToDecrypt, COOKIE_ENCRYPTION_KEY);
    
    return JSON.parse(bytes.toString(encoder) ?? '{}');
  } catch (error) {
    console.error(error);
    return value;
  }
}
```

Reply:
Well - it's just what if I want to store a string?

My code will let you store a string if you want, but if you sent an object, you'll get one back..

Reply:
You can `JSON.parse()` a string too...

```typescript
const foo = JSON.stringify('foo'); // '"foo"';

JSON.parse(foo); // 'foo'
```

Reply:
I've had to do the same for the `localStorage` and `sessionStorage` drivers.